### PR TITLE
[CLEANUP] Case cleanup cpp and proto files refactor (depends on #109)

### PR DESCRIPTION
**Pull request by AntoineRondelet**

### CODE REVIEW

**Comment by AntoineRondelet** [comment]
I haven't modified the python client to reflect the changes made in the proto files, so the python tests will fail for now

File: api/groth16_messages.proto [hunk 1]
```diff
@@ -1,17 +1,17 @@
 syntax = "proto3";
 
-package proverpkg;
+package prover_proto;
 
 import public "api/util.proto";
 
-message R1csGgPpzksnarkVerificationKey {
+message VerificationKeyGROTH16 {
     HexadecimalPointBaseGroup1Affine alpha_g1 = 1;
```
Comment (dtebbs) [inline api/groth16_messages.proto]: Just a thought, but while doing this big rename, how about renaming these to `HexPoint....`?  It shortens things without adding any ambiguity.
Also, out of curiosity, what does `Base` refer to here?

File: src/types/joinsplit.hpp [hunk 2]
```diff
@@ -10,26 +10,28 @@
 namespace libzeth
 {
 
-// We simplify the interface of the JSInput object compared to what Zcash did
-// In fact, all our fields are going to be computed from another component
-// written in python or js, that will use the ethereum primitives to hash.
-template<typename FieldT> class JSInput
+// We simplify the interface of the joinsplit_input object compared to what
+// Zcash did In fact, all our fields are going to be computed from another
```
[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply src/types/joinsplit.hpp]: Exact, thanks

File: src/types/joinsplit.hpp [hunk 3]
```diff
@@ -10,26 +10,28 @@
...
+// hash.
+template<typename FieldT> class joinsplit_input
 {
 public:
     // --- Merkle tree witness (path, and address)
     std::vector<FieldT> witness_merkle_path;
-    bitsAddr address_bits; // boolean vector of length the depth of the merkle
-                           // tree containing the binary encoding of the address
+    bits_addr
+        address_bits; // boolean vector of length the depth of the merkle
```
[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply src/types/joinsplit.hpp]: Exact, thanks

File: src/snarks/groth16/helpers.tcc [hunk 4]
```diff
@@ -133,23 +152,26 @@ void proofAndInputToJson(
...
-    ss << "{\n";
-    ss << " \"a\" :[" << outputPointG1AffineAsHex(proof.g_A) << "],\n";
-    ss << " \"b\"  :[" << outputPointG2AffineAsHex(proof.g_B) << "],\n";
-    ss << " \"c\" :[" << outputPointG1AffineAsHex(proof.g_C) << "],\n";
-    ss << " \"input\" :"
-       << "["; // 1 should always be the first variable passed
-
+    ss << "{"
+       << "\n"
+       << "\t\"a\" :[" << get_point_g1_affine_as_hex_str(proof.g_A) << "],"
```
[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply src/snarks/groth16/helpers.tcc]: `point_g1_affine_as_hex` looks good to me yes

File: api/groth16_messages.proto (hunk 1 above)
Comment (AntoineRondelet) [reply api/groth16_messages.proto]: `HexPoint....` that works for me yes.

> Also, out of curiosity, what does Base refer to here?
A bilin. pairing is denoted as e: G1 x G2 -> Gt, where both G1 and G2 are the base groups, and where Gt is the target group

**Comment by AntoineRondelet** [comment]
LGTM

**Pull request merged**